File: Code/new_fitervari_backend/src/main/java/at/htl/model/Set.java
package at.htl.model;

import javax.persistence.*;

@Entity
public class Set {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int repetitions;
    private double distance;
    private double weight;
    private double time;
    @Column(name = "set_number")
    private int setNumber;
    private String type;

    //region GetterSetter
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public int getRepetitions() {
        return repetitions;
    }

    public void setRepetitions(int repetitions) {
        this.repetitions = repetitions;
    }

    public double getDistance() {
        return distance;
    }

    public void setDistance(double distance) {
        this.distance = distance;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public double getTime() {
        return time;
    }

    public void setTime(double time) {
        this.time = time;
    }

    public int getSetNumber() {
        return setNumber;
    }

    public void setSetNumber(int setNumber) {
        this.setNumber = setNumber;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
    //endregion
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/Set.kt
package at.htl_leonding.model

import io.quarkus.hibernate.orm.panache.PanacheEntity
import javax.persistence.Column
import javax.persistence.Entity

@Entity
class Set(
        var repetitions: Int,
        var distance: Double,
        var weight: Double,
        var time: Double,
        @Column(name = "set_number")
        var setNumber: Double,
        var type: String
) : PanacheEntity() {
    fun copyValues(other: Set) {
        this.repetitions = other.repetitions;
        this.distance = other.distance;
        this.weight = other.weight;
        this.time = other.time;
        this.setNumber = other.setNumber;
        this.type = other.type;
    }
}